src/main/java/io/github/syst3ms/skriptparser/registration/ContextValueManager.java

```diff
@@ -0,0 +1,111 @@
+     */
+    public static class ContextValue<T> {
+        private final Class<? extends TriggerContext> context;
+        private final Class<T> type;
+        private final String name;
+        private final Function<TriggerContext, T[]> contextFunction;
+        private final int timeline;
+
+        public static int PAST = -1;
+        public static int PRESENT = 0;
+        public static int FUTURE = 1;
+
+        /**
+         * Construct a context value.
+         *
+         * @param context         the specific {@link TriggerContext} class
+         * @param name            the suffix of this value
+         * @param contextFunction the function to apply to the context
+         */
+        public ContextValue(Class<? extends TriggerContext> context, Class<T> type, String name, Function<TriggerContext, T[]> contextFunction) {
+            this(context, type, name, contextFunction, 0);
+        }
+
+        /**
+         * Construct a context value.
+         *
+         * @param context         the specific {@link TriggerContext} class
+         * @param name            the suffix of this value
+         * @param contextFunction the function to apply to the context
+         * @param timeline        whether this value represent a present, past or future state
+         */
+        public ContextValue(Class<? extends TriggerContext> context, Class<T> type, String name, Function<TriggerContext, T[]> contextFunction, int timeline) {
+            this.context = context;
+            this.type = type;
+            this.name = name.toLowerCase();
+            this.contextFunction = contextFunction;
+            this.timeline = normalTimeline(timeline);
+        }
+
+        public Class<? extends TriggerContext> getContext() {
+            return context;
+        }
+
+        /**
+         * @return the returned type of this context value
+         */
+        public Class<T> getType() {
+            return type;
+        }
+
+        /**
+         * Returns the name of the context value.
+         * If the name, for example, is 'test', the use case will be 'context-test'
+         * @return the name of this context value
+         */
+        public String getName() {
+            return name;
+        }
+
+        /**
+         * @return the function that needs to be applied in order to get the context value
+         */
+        public Function<TriggerContext, T[]> getContextFunction() {
+            return contextFunction;
+        }
+
+        /**
+         * Returns {@code -1} for the past, {@code 0} for the present and {@code 1} for the future.
+         * @return whether this happens in the present, past or future
+         */
+        public int getTimeline() {
+            return timeline;
+        }
+
+        public boolean matches(Class<? extends TriggerContext> handledContext, String name, int timeline) {
+            return handledContext.equals(this.context)
+                    && this.name.equals(name)
```

Comment:
`this.name` is always lowercase, but `name` isn't guaranteed to be. I think toLowerCase() isn't necessary for `this.name`, and instead this line should use `equalsIgnoreCase`.